contracts/src/token/erc1155/extensions/metadata_uri.rs

```diff
@@ -0,0 +1,101 @@
+//! Optional URI Metadata of the ERC-1155 standard, as defined
+//! in the [ERC].
+//!
+//! [ERC]: https://eips.ethereum.org/EIPS/eip-1155#metadata-extensions
+
+use alloc::string::String;
+
+use alloy_primitives::{FixedBytes, U256};
+use alloy_sol_macro::sol;
+use openzeppelin_stylus_proc::interface_id;
+use stylus_sdk::stylus_proc::{public, sol_storage};
+
+use crate::utils::introspection::erc165::{Erc165, IErc165};
+
+sol! {
+    /// Emitted when the URI for token type `id` changes to `value`, if it is
+    /// a non-programmatic URI.
+    ///
+    /// If a [`URI`] event was emitted for `id`, the standard [guarantees] that
+    /// `value` will equal the value returned by [`IErc1155MetadataUri::uri`].
+    ///
+    /// [guarantees]: https://eips.ethereum.org/EIPS/eip-1155#metadata-extensions
+    #[allow(missing_docs)]
+    event URI(string value, uint256 indexed id);
+}
+
+sol_storage! {
+    /// URI Metadata of an [`crate::token::erc1155::Erc1155`] token.
+    pub struct Erc1155MetadataUri {
+        /// Used as the URI for all token types by relying on ID substitution,
+        /// e.g. https://token-cdn-domain/{id}.json.
+        string _uri;
+    }
+}
+
+/// Interface for the optional metadata functions from the ERC-1155 standard.
+#[interface_id]
+pub trait IErc1155MetadataUri {
+    /// Returns the URI for token type `id`.
+    ///
+    /// If the `\{id\}` substring is present in the URI, it must be replaced by
+    /// clients with the actual token type ID.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `id` - Token id.
+    fn uri(&self, id: U256) -> String;
+}
+
+// FIXME: Apply multi-level inheritance to export Metadata's functions.
```

Comment:
Do we need this comment here? I am not sure as we do not re-export anything here.

Reply:
Then it makes sense to delete [the same comment from IErc721Metadata](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/feat/erc1155-metadata/contracts/src/token/erc721/extensions/metadata.rs#L42)?

Reply:
@0xNeshi there is other case in `Erc721Metadata`, as it derives from `Metadata`.